ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/QueryCompactor.java

```diff
@@ -272,5 +277,23 @@ static void removeFilesForMmTable(HiveConf conf, AcidDirectory dir) throws IOExc
         fs.delete(dead, true);
       }
     }
+
+    static void overrideConfProps(HiveConf conf, CompactionInfo ci, Map<String, String> properties) {
+      for (String key : properties.keySet()) {
+        if (key.startsWith(COMPACTOR_PREFIX)) {
+          String property = key.substring(10); // 10 is the length of "compactor." We only keep the rest.
+          conf.set(property, properties.get(key));
+        }
+      }
+
+      // Give preference to properties coming from compaction
+      // over table properties
+      for (String key : ci.getPropertiesMap().keySet()) {
```

Comment:
why is this required? those properties shouldn't affect the execution

Reply:
This is required for the `ALTER TABLE <tablename> COMPACT 'major' WITH TBLPROPERTIES OVERWRITE (<property list>)` scenario. Here the properties are entered into `COMPACTION_QUEUE` and when the table is picked for compaction, the table properties which are set in `CQ_TBLPROPERTIES` must also be picked. All this information is stored in CompactionInfo, hence the need to set the properties as here.

Same behaviour is seen for MR-based compaction as well.

Reply:
could we deduplicate this part of the code? something like:
````
static void overrideConfProps(HiveConf conf, CompactionInfo ci, Map<String, String> properties) {
      Stream.of(properties, new StringableMap(ci.properties))
        .filter(Objects::nonNull)
        .flatMap(map -> map.entrySet().stream())
        .filter(entry -> entry.getKey().startsWith(COMPACTOR_PREFIX))
        .forEach(entry -> {
          String property = entry.getKey().substring(COMPACTOR_PREFIX.length());
          conf.set(property, entry.getValue());
        });
    }
````

Reply:
Done.